### PR TITLE
fix node shutdown/reboot tests MQ-2524

### PR DESCRIPTION
**Pull request by payes**
This PR achieves the following:
1. Add node states Unknown and Empty
2. Update error check string for REST errors to control planes
3. Add exclusive list of non control plane components
4. Use msn interface for node reboot/shutdown tests

Signed-off-by: Payes Anand <payes.anand@mayadata.io>

### CODE REVIEW

File: src/common/k8stest/util_testpods.go [hunk 1]
```diff
@@ -426,7 +429,8 @@ func RestartMayastorPods(timeoutSecs int) error {
 						continue
 					}
 					for _, newPodName := range newPodNames {
-						found = found || prevPodName == newPodName
+						// Avoiding checking for nats here as nats is an sts
```
Comment (chriswldenyer) [inline src/common/k8stest/util_testpods.go:432]: nats pods also do not change their names.
Does this mean they can be treated like mayastor-etcd, i.e. can be ignored?

File: src/node_failure/node_failure_test.go [hunk 2]
```diff
@@ -184,20 +183,23 @@ func (c *failureConfig) RebootDesiredNodes(uuid string) {
...
 
-func (c *failureConfig) verifyMayastorComponentStates() {
-	Eventually(func() bool {
-		nodeList, err := custom_resources.ListMsNodes()
-		Expect(err).ToNot(HaveOccurred(), "ListMsNodes")
-		for _, node := range nodeList {
-			if node.Status != "online" {
-				return false
-			}
+func (c *failureConfig) verifyMayastorComponentStates(numMayastorInstances int) {
```
Comment (chriswldenyer) [inline src/node_failure/node_failure_test.go:186]: So it no longer waits for the nodes to be online. Presumably intentional?

**Review by chriswldenyer — APPROVED** [review]
LGTM, some comments.

File: src/common/k8stest/util_testpods.go (hunk 1 above)
Comment (blaisedias) [reply src/common/k8stest/util_testpods.go:432]: The restart logic is suspect - I have a version which replaces the name change dependency with a time dependency which can then be used to check daemonsets, deployments and statefulsets.
For now restart checking only works if pod names change :-(

**Review by blaisedias — COMMENTED** [review]
Change looks okay to me. But I would be happier if you added more description along the lines of what your trying to achieve to the git commit and by implication the github PR.
Please reference the relevant CAS.

File: src/node_failure/node_failure_test.go (hunk 2 above)
Comment (payes) [reply src/node_failure/node_failure_test.go:186]: Hi @chriswldenyer , It is still there:
	nodeList, err := k8stest.ListMsns()
	Expect(err).ToNot(HaveOccurred(), "ListMsNodes")
	count := 0
	for _, node := range nodeList {
		status, err := k8stest.GetMsNodeStatus(node.Name)
		Expect(err).ToNot(HaveOccurred(), "GetMsNodeStatus")
		if status == controlplane.NodeStateOnline() {
			count++
		}
	}

Comment (payes) [reply src/node_failure/node_failure_test.go:186]: Hi @chriswldenyer , It is still there:
nodeList, err := k8stest.ListMsns()
Expect(err).ToNot(HaveOccurred(), "ListMsNodes")
count := 0
for _, node := range nodeList {
status, err := k8stest.GetMsNodeStatus(node.Name)
Expect(err).ToNot(HaveOccurred(), "GetMsNodeStatus")
if status == controlplane.NodeStateOnline() {
count++
}
}

Comment (payes) [reply src/node_failure/node_failure_test.go:186]: Hi @chriswldenyer , It is still there:
nodeList, err := k8stest.ListMsns()
Expect(err).ToNot(HaveOccurred(), "ListMsNodes")
count := 0
for _, node := range nodeList {
status, err := k8stest.GetMsNodeStatus(node.Name)
Expect(err).ToNot(HaveOccurred(), "GetMsNodeStatus")
if status == controlplane.NodeStateOnline() {
count++
}
}

File: src/common/k8stest/util_testpods.go (hunk 1 above)
Comment (payes) [reply src/common/k8stest/util_testpods.go:432]: Sure, I will remove this part from this PR then.

**Comment by payes** [comment]
> Change looks okay to me. But I would be happier if you added more description along the lines of what your trying to achieve to the git commit and by implication the github PR. Please reference the relevant CAS.

Done

**Review by blaisedias — APPROVED** [review]

**Comment by payes** [comment]
bors merge

**Pull request merged**